backend/models/ocr.js

```diff
@@ -1,8 +1,124 @@
-//THIS IS A STUB TO TEST INTEGRATION
+var request = require('request');
+var ocrKey = process.env.OCR_KEY;
+var spotify = require('../models/spotifyPlaylist.js');
+
+var requestForm = {
+  url: 'https://api.ocr.space/parse/image',
+  form: {
+    apikey: ocrKey,
+    isOverlayRequired: true
+  }
+};
+
 exports.parseImage = function(url) {
-  return new Promise(function(resolve,reject){
-    setTimeout(function(){
-      resolve(["beyonce","coldplay"]);
-    },500);
+  return parseImagetoBlob(url)
+    .then(function(blob) {
+      var firstGroup = blob.slice(0,10);
+      return searchForAllArtists(firstGroup)
+        .then(function(list) {
+          return list.map(function(val) {
+            return val.name;
+          });
+        });
+    });
+};
+
+parseImagetoBlob = function(url) {
+  return sendUrlToOcr(url)
+    .then(parseLines);
+};
+
+searchForAllArtists = function(data){
+  return Promise.all(data.map(function(value, index, array){
+    return searchAPIForArtist(value.words);
+  }));
+};
+
+searchAPIForArtist = function(array){
+  if(array.length < 1){
+      return null;
+    }
+  var possibleArtistString = buildString(array);
+  return searchSpotifyForArtist(possibleArtistString)
+    .then(function(result){
+      if(result.length){
+        return result[0];
+      }else{
+        return searchAPIForArtist(array.slice(0, -1));
+      }
+    });
+};
+
+buildString = function(array){
+  return array.join(" ");
+};
+
+buildTestString = function(array){
+  return array.map(function(value, index, array){
+    return array.slice(0, index+1).join(" ");
+  }).reverse();
+};
+
+searchSpotifyForArtist = function(query) {
+  return spotify.searchForArtist(query);
+};
+
+
+
+filterDate = function(blob) {
+  blob.some(function(val,ind,arr) {
+    if(containsDate(val)) {
+      var possibleDate = arr.splice(ind,1);
+      return true;
+    }
+  });
+};
+
+containsDate = function(sentence) {
+  return sentence.words.some(function(value) {
+    return DATES.includes(value);
   });
 };
+
+sendUrlToOcr = function(url) {
+  return new Promise(function(resolve,reject) {
+    requestForm.form.url = url;
+    request.post(requestForm, function(err, data) {
+      resolve(data.body);
+    });
+  });
+};
+
+parseLines = function(inputJSON) {
+  return JSON.parse(inputJSON)
+    .ParsedResults[0].TextOverlay.Lines
+    .map(function(entry){
+      var wordList = entry.Words.map(function(word) {
+        return word.WordText;
```

Comment:
To elaborate further, many of the functions are hacked together to be able to test the rest of the product, so refactoring all this stuff is on the list right now